packages/plugin-ext/src/main/browser/webviews-main.ts

```diff
@@ -98,7 +98,14 @@ export class WebviewsMainImpl implements WebviewsMain {
         }
     }
     $reveal(handle: string, showOptions: WebviewPanelShowOptions): void {
-        throw new Error('Method not implemented.');
+        const webview = this.getWebview(handle);
+        if (webview.isDisposed) {
+            return;
+        }
+        this.shell.revealWidget(webview.id);
+        if (showOptions.preserveFocus) {
+            webview.focus();
```

Comment:
`this.shell.activateWidget(widget)` - in order to tell the shell to bring a widget to focus, `webview` has to call `focus` in  `onActivateRequest` callback

Reply:
also It looks bogus, `preserverFocus` means that a web view should not get focus

```ts
if (showOptions.preserveFocus) {
    shell.activateWidget(webview.id);
} else {
    shell.revealWidget(webview.id);
}
```

and then call  `this.node.focus()` in `onActivateRequest` callback on webview widget

I will modify sample extension to test it.

Reply:
ok thanks